File: modules/services/airflow/dags/lectorium/tracks/models/track.py
from typing import TypedDict


class TrackLanguage(TypedDict):
  language: str
  source: str # track | transcript
  type: str # original | generated

class TrackTranscript(TypedDict):
  path: str

class TrackAudio(TypedDict):
  path: str
  file_size: int
  duration: int

class Track(TypedDict):
  _id: str
  type: str
  version: int
  location: str | None
  date: list[int] | None
  author: str
  title: dict[str, str]
  references: list[list[str|int]]
  audio: dict[str, TrackAudio]
  transcripts: dict[str, TrackTranscript]
  languages: list[TrackLanguage]
  tags: list[str] | None

File: modules/libs/dal/models/Track.ts
export type Track = {
  _id: string;
  type: "track";
  version: number;
  location: string;
  date: [number, number, number];
  author: string;
  title: Record<string, string>;
  references: Array<string|number>[];
  audio: {
    original: {
      path: string;
      fileSize: number;
      duration: number;
    };
  };
  languages: {
    language: string;
    source: "track" | "transcript";
    type: "original" | "generated" | "edited";
  }[];
  transcripts: Record<string, {
    path: string;
  }>;
  tags?: string[];
}
